src/Engine/babylon.engine.ts

```diff
@@ -1784,37 +1784,58 @@
             return this._vrDisplay;
         }
 
+        /**
+         * Initializes a webVR display and starts listening to display change events.
+         * The onVRDisplayChangedObservable will be notified upon these changes.
+         * @returns The onVRDisplayChangedObservable.
+         */
         public initWebVR(): Observable<{ vrDisplay: any, vrSupported: any }> {
-            var notifyObservers = () => {
-                var eventArgs = {
-                    vrDisplay: this._vrDisplay,
-                    vrSupported: this._vrSupported
-                };
-                this.onVRDisplayChangedObservable.notifyObservers(eventArgs);
-            }
+            this.initWebVRAsync();
+            return this.onVRDisplayChangedObservable;
+        }
 
-            if (!this._onVrDisplayConnect) {
-                this._onVrDisplayConnect = (event) => {
-                    this._vrDisplay = event.display;
-                    notifyObservers();
-                };
-                this._onVrDisplayDisconnect = () => {
-                    this._vrDisplay.cancelAnimationFrame(this._frameHandler);
-                    this._vrDisplay = undefined;
-                    this._frameHandler = Tools.QueueNewFrame(this._bindedRenderFunction);
-                    notifyObservers();
-                };
-                this._onVrDisplayPresentChange = () => {
-                    this._vrExclusivePointerMode = this._vrDisplay && this._vrDisplay.isPresenting;
+        /**
+         * Initializes a webVR display and starts listening to display change events.
+         * The onVRDisplayChangedObservable will be notified upon these changes.
+         * @returns A promise containing a VRDisplay and if vr is supported.
+         */
+        public initWebVRAsync(): Promise<{ vrDisplay: Nullable<any>, vrSupported: boolean }> {

```

Comment:
if you store the return-value of this function (for example engine.webVRInitPromise) you could always "then" it. After calling initWebVRAsync, you could use `engine.webVRInitPromise.then(() => {})` anywhere in your code. This way you make sure not to init webvr twice. you can also return this promise from this function if it already exists. Something along the line of:
```javascript
this.webVRInitPromise = this.webVRInitPromise || new Promise.....
return this.webVRInitPromise
```

Reply:
I'm not sure doing this is desired for situations when headsets are disconnected and connected so Ill leave this out for now.

Reply:
This should be controlled by the engine. In that case, you can store a rejected promise in this variable, and the developer will still be able to get the right state.

Reply:
Did I do this as expected in this commit? https://github.com/BabylonJS/Babylon.js/pull/3693/commits/7763d07f85d6d6e772877f17c7c45169f990f838